storage/storage.go

```diff
@@ -999,6 +1000,15 @@ type ObjectAttrsToUpdate struct {
 	// If not empty, applies a predefined set of access controls. ACL must be nil.
 	// See https://cloud.google.com/storage/docs/json_api/v1/objects/patch.
 	PredefinedACL string
+
+	// Retention contains the retention configuration for this object.
+	Retention *ObjectRetention
+
+	// OverrideUnlockedRetention provides an option for overriding an Unlocked
+	// Retention policy. This must be set to true in order to change a policy
+	// from Unlocked to Locked, to set it to null, or to reduce its
+	// retainUntilTime attribute.
```

Comment:
I think this could be a little clearer; is sending Retention and not OverrideUnlockedRetention ever valid? It's not obvious to me from reading this.

Overall having this field still seems quite clunky to me; but we can discuss offline and/or get a consult from the product team. If we really want to include it, it seems best to have it as a method on ObjectHandle as we do with other query params to the JSON API.

Reply:
As discussed offline, since it's a permanent change that can have big consequences (not being able to delete the object, or allowing the object to be deleted before intended), we are required to provide an extra level of "are you sure?" for certain changes to the Retention. 

I believe we do have other query params as fields on the update attrs, but in this case it is just a param that we send and do not report back in any way, so you're right, it makes sense to have this as a method. I updated this.